### ISSUE TITLE
Encountered channel not found error on adding Windows integration to the Windows agent.

### ISSUE THREAD

**Issue by amolnater-qasource** [elastic/elastic-agent#5746]
**Kibana Build details:**
```
VERSION: 8.16.0 SNAPSHOT
BUILD: 78938
COMMIT: 7b832691e8b07c67b411da95b0398a04711da864
```
Artifact: https://snapshots.elastic.co/8.16.0-39df64b4/downloads/beats/elastic-agent/elastic-agent-8.16.0-SNAPSHOT-windows-x86_64.zip

![Image](https://github.com/user-attachments/assets/1bba0dcd-7155-43b8-aa47-bc1ea7587e80)

Host: Windows Server 2022- Test Signing ON

**Preconditions:**
1. 8.16.0 SNAPSHOT Kibana cloud environment should be available.
2. Agent should be installed with policy having System and Windowsintegrations.

**Steps to reproduce:**
1. Navigate to Agents tab.
2. Observe the Agent gets unhealthy and navigate to policy details page.
3. Observe error for Windows integration: `Encountered channel not found error`

**Expected Result:**
No error should be displayed on adding Windows integration to the Windows agent.

**Logs:**
[elastic-agent-diagnostics-2024-10-09T06-48-15Z-00.zip](https://github.com/user-attachments/files/17304782/elastic-agent-diagnostics-2024-10-09T06-48-15Z-00.zip)

**Screenshots:**
![Image](https://github.com/user-attachments/assets/f7753844-5fab-4990-b6df-17158e70fa0c)
![Image](https://github.com/user-attachments/assets/bcd33967-f836-4e13-8dac-b0070a2903b1)

**Comment by amolnater-qasource** [comment]
@muskangulati-qasource Please review.

**Comment by muskangulati-qasource** [comment]
Secondary review is done for this ticket!

**Comment by cmacknz** [comment]
I see this is privileged/admin agent looking in agent-info.yaml:

```yaml
agent_id: 881c5687-32af-4bf9-b62f-4b74f2f688ec
headers: {}
log_level: info
snapshot: true
unprivileged: false
version: 8.16.0

```

Also that this is coming from the winlog input. Tagging @nfritts and @elastic/sec-windows-platform.

```yaml
            input-winlog-default-winlog-windows-4ea5f67a-48fc-41ea-b586-2a29eac6423a:
                message: 'Encountered channel not found error when opening Windows Event Log: The specified channel could not be found.'
                payload:
                    streams:
                        winlog-windows.forwarded-4ea5f67a-48fc-41ea-b586-2a29eac6423a:
                            error: ""
                            status: HEALTHY
                        winlog-windows.powershell-4ea5f67a-48fc-41ea-b586-2a29eac6423a:
                            error: ""
                            status: HEALTHY
                        winlog-windows.powershell_operational-4ea5f67a-48fc-41ea-b586-2a29eac6423a:
                            error: ""
                            status: HEALTHY
                        winlog-windows.sysmon_operational-4ea5f67a-48fc-41ea-b586-2a29eac6423a:
                            error: 'Encountered channel not found error when opening Windows Event Log: The specified channel could not be found.'
                            status: DEGRADED
```

**Comment by bjmcnic** [comment]
This issue has arisen following changes in this PR (https://github.com/elastic/beats/pull/40163).  

The default configuration for for the Windows Integration has historically included Sysmon Operational channel.  Sysmon is not a core component of Windows, it's a SysInternals tool (https://learn.microsoft.com/en-us/sysinternals/downloads/sysmon) that users can download and use at their discretion.  As such, the Windows Integration has historically failed to open the Sysmon Operational channel, but that didn't propagate a `DEGRADED` status until the recent PR.

Users can remedy the degraded status by either installing Sysmon, causing the channel to exist; or by deselecting the Sysmon Operational channel in the Windows Integration configuration...

![Image](https://github.com/user-attachments/assets/4aaa5344-f006-438e-9a2b-759d0818b7dd)

Possible solutions:
- Revert the code change that causes a `DEGRADED` status when a configured channel is not found. 
  - Obviously that could risk missing a helpful `DEGRADED` status that could indicated an unexpectedly missing non-default channel.
- Document and explain how this is now more correct.
- Change the Windows Integration default to not include Sysmon.

Thoughts @cmacknz @nfritts @andrewkroh ?

**Comment by jamiehynds** [comment]
In an ideal world, we'd move Sysmon out of Windows and have it as a standalone integration but that'd be very disruptive for existing users and would likely impact rules, dashboards, etc. 

As a quick fix, could we exclude Sysmon from our `DEGRADED` logic? So keep Sysmon within the Windows integration, but if a user doesn't have Sysmon installed, we don't trigger a `DEGRADED` status?

**Comment by intxgo** [comment]
Can't Agent get it from `https://live.sysinternals.com/Sysmon64.exe` and install if it's missing? When adding policy with sysmon data collection.

**Comment by bjmcnic** [comment]
@jamiehynds 

> As a quick fix, could we exclude Sysmon from our DEGRADED logic? So keep Sysmon within the Windows integration, but if a user doesn't have Sysmon installed, we don't trigger a DEGRADED status?

I don't see a technical reason we couldn't just insert a check for whether we're trying to grab that particular channel near the code that changed.  But that's filebeat code and not the Windows integration code.   That'd be kind of an awkward place for the check in the long term and wouldn't scale well if we want to handle other things differently.

I noticed the PR that changed this was addressing: https://github.com/elastic/beats/issues/39735.  Which related to wanting to see failures for channels when permission is denied, typically when Agent is installed unprivileged.  I recreated that and now DO see the desired `Access is denied` error:

```
c:\>"c:\Program Files\Elastic\Agent\elastic-agent.exe" status
┌─ fleet
│  └─ status: (HEALTHY) Connected
└─ elastic-agent
   ├─ status: (DEGRADED) 1 or more components/units in a failed state
   ├─ system/metrics-default
   │  ├─ status: (HEALTHY) Healthy: communicating with pid '8300'
   │  └─ system/metrics-default-system/metrics-system-68951c20-696b-4518-b64c-63de7317ef29
   │     └─ status: (DEGRADED) Error fetching data for metricset system.diskio: disk io counters: cannot open new key in the registry in order to enable the performance counters: Access is denied.
   ├─ windows/metrics-default
   │  ├─ status: (HEALTHY) Healthy: communicating with pid '5976'
   │  └─ windows/metrics-default-windows/metrics-windows-e02b045d-e899-4ef1-b6ce-400be3d94119
   │     └─ status: (FAILED) 1 error: initialization of reader failed: failed to expand counter (query='\Process(*)\% Processor Time'): Unable to connect to the specified computer or the computer is offline.
   └─ winlog-default
      ├─ status: (HEALTHY) Healthy: communicating with pid '5416'
      ├─ winlog-default-winlog-system-68951c20-696b-4518-b64c-63de7317ef29
      │  └─ status: (DEGRADED) failed to open Windows Event Log channel "Security": Access is denied.
      └─ winlog-default-winlog-windows-e02b045d-e899-4ef1-b6ce-400be3d94119
         └─ status: (DEGRADED) Encountered channel not found error when opening Windows Event Log: The specified channel could not be found.
```

The thing that strikes me there is we have multiple types of failures for `winlog-default`.  One we newly want reported that hadn't been, and one we don't want reported that previously hadn't been.

I wonder if a good long term solution would be for the integration to feed in some type of filter data along with each channel it wants.  Some enum or struct that tells it to actually degrade on access denied errors for this channel, but ignore not found errors for this channel, or warn/log (but not `DEGRADE`) for some other type of error for some other channel.  Such that the integration can communicate how specific channel subscription failures should impact the integration's state.  Trying to place that logic in code in filebeat seems awkward.

I'm not sure that change could be ready for 8.16.0.  Perhaps we should rollback the change that caused this issue to arise and continue to tolerate the missing `Access is denied` as unprivileged appears to still be technically beta.  And then we could incorporate the channel specific failure handling from the Windows integration in the next release.  Thoughts? @cmacknz @jamiehynds @nfritts

**Comment by cmacknz** [comment]
Having a per input way to turn off the "errors mark the Beat as degraded" would make sense vs just reverting the entire feature. This config could later expand into a list of specific errors to mute. For system/metrics we had similar ideas in https://github.com/elastic/beats/issues/40543 but it hasn't been implemented yet. 

We have been more focused on fixing the specific errors, which in many cases have been actual bugs or permissions errors we were handling improperly. For system/metrics we also only get these errors when unprivileged. 

The OTel collector process scraper allows muting specific categories of error which is what we'd eventually want to emulate https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/hostmetricsreceiver/README.md#process

**Comment by cmacknz** [comment]
@bjmcnic followed up separately and we agreed we should revert the winlog specific change here in 8.16 while we work on a proper fix for 8.17 considering that:
1. This affects privileged agents for winlog in the default configuration AKA everybody on windows.
2. The final 8.16.0 BC is Thursday
3. The linked PR is specific to the winlog input

There's not time to do a more in depth fix and the current state will probably lead to a flood of support cases. The revert PR is in https://github.com/elastic/beats/pull/41468

**Comment by amolnater-qasource** [comment]
Hi Team,

While testing on 8.17.0 SNAPSHOT, we have found this issue reproducible there too.

**Observations:**
- Encountered channel not found error on adding Windows integration to the Windows agent.

**Build details:**
VERSION: 8.17.0 SNAPSHOT
BUILD: 80188
COMMIT: fdb16ae8cbdf4236db3696aa00d0bb98c943d864
Artifact Link: https://snapshots.elastic.co/8.17.0-7a041bf5/downloads/beats/elastic-agent/elastic-agent-8.17.0-SNAPSHOT-windows-x86_64.zip

![Image](https://github.com/user-attachments/assets/ee34c3ff-4872-4085-974d-e423405c187a)

**Screenshot:**
![Image](https://github.com/user-attachments/assets/477bbe01-520f-4f7d-a531-a878e05e3378)

**Logs:**
[elastic-agent-diagnostics-2024-11-18T09-03-35Z-00.zip](https://github.com/user-attachments/files/17797546/elastic-agent-diagnostics-2024-11-18T09-03-35Z-00.zip)

Please let us know if us know if anything else is required from our end.

Thanks!

**Comment by bjmcnic** [comment]
It's fixed in `8.16.0`.  Looks like the revert of the change was to `8.16` branch, but hasn't hit `main`.

```
c:\>"c:\Program Files\Elastic\Agent\elastic-agent.exe" version
Binary: 8.16.0 (build: 3f07f2fd932f20e972399306d394763ade6b74b4 at 2024-11-07 13:33:43 +0000 UTC)
Daemon: 8.16.0 (build: 3f07f2fd932f20e972399306d394763ade6b74b4 at 2024-11-07 13:33:43 +0000 UTC)

c:\>"c:\Program Files\Elastic\Agent\elastic-agent.exe" status --output full
┌─ fleet
│  └─ status: (HEALTHY) Connected
└─ elastic-agent
   ├─ status: (HEALTHY) Running
   ├─ info
   │  ├─ id: 4eb139dd-aa42-4ec4-9463-39f1b7b37f60
   │  ├─ version: 8.16.0
   │  └─ commit: 3f07f2fd932f20e972399306d394763ade6b74b4
   ├─ beat/metrics-monitoring
   │  ├─ status: (HEALTHY) Healthy: communicating with pid '5128'
   │  ├─ beat/metrics-monitoring
   │  │  ├─ status: (HEALTHY) Healthy
   │  │  └─ type: OUTPUT
   │  └─ beat/metrics-monitoring-metrics-monitoring-beats
   │     ├─ status: (HEALTHY) Healthy
   │     └─ type: INPUT
   ├─ filestream-monitoring
   │  ├─ status: (HEALTHY) Healthy: communicating with pid '520'
   │  ├─ filestream-monitoring
   │  │  ├─ status: (HEALTHY) Healthy
   │  │  └─ type: OUTPUT
   │  └─ filestream-monitoring-filestream-monitoring-agent
   │     ├─ status: (HEALTHY) Healthy
   │     └─ type: INPUT
   ├─ http/metrics-monitoring
   │  ├─ status: (HEALTHY) Healthy: communicating with pid '3876'
   │  ├─ http/metrics-monitoring
   │  │  ├─ status: (HEALTHY) Healthy
   │  │  └─ type: OUTPUT
   │  └─ http/metrics-monitoring-metrics-monitoring-agent
   │     ├─ status: (HEALTHY) Healthy
   │     └─ type: INPUT
   ├─ log-default
   │  ├─ status: (HEALTHY) Healthy: communicating with pid '5944'
   │  ├─ log-default
   │  │  ├─ status: (HEALTHY) Healthy
   │  │  └─ type: OUTPUT
   │  └─ log-default-logfile-system-f57fba90-d162-43bb-8f6b-546015d84c78
   │     ├─ status: (HEALTHY) Healthy
   │     └─ type: INPUT
   ├─ system/metrics-default
   │  ├─ status: (HEALTHY) Healthy: communicating with pid '3500'
   │  ├─ system/metrics-default
   │  │  ├─ status: (HEALTHY) Healthy
   │  │  └─ type: OUTPUT
   │  └─ system/metrics-default-system/metrics-system-f57fba90-d162-43bb-8f6b-546015d84c78
   │     ├─ status: (HEALTHY) Healthy
   │     └─ type: INPUT
   ├─ windows/metrics-default
   │  ├─ status: (HEALTHY) Healthy: communicating with pid '2220'
   │  ├─ windows/metrics-default
   │  │  ├─ status: (HEALTHY) Healthy
   │  │  └─ type: OUTPUT
   │  └─ windows/metrics-default-windows/metrics-windows-e715bcd7-f597-4666-9a17-c75be66c9e02
   │     ├─ status: (HEALTHY) Healthy
   │     └─ type: INPUT
   └─ winlog-default
      ├─ status: (HEALTHY) Healthy: communicating with pid '4692'
      ├─ winlog-default
      │  ├─ status: (HEALTHY) Healthy
      │  └─ type: OUTPUT
      ├─ winlog-default-winlog-system-f57fba90-d162-43bb-8f6b-546015d84c78
      │  ├─ status: (HEALTHY) Healthy
      │  └─ type: INPUT
      └─ winlog-default-winlog-windows-e715bcd7-f597-4666-9a17-c75be66c9e02
         ├─ status: (HEALTHY) Healthy
         └─ type: INPUT

c:\>
```

**Comment by amolnater-qasource** [comment]
Hi Team,
We have revalidated this issue on latest 8.17.0 BC5 kibana cloud environment and found it **fixed now**.

**Observations:**
- No errors are displayed on adding Windows integration to the Windows agent and agent remains Healthy throughout.

**Screenshots:**
![Image](https://github.com/user-attachments/assets/1fad6a98-e562-4a6e-9c0d-ea96fc948998)
![Image](https://github.com/user-attachments/assets/32d6fa62-ff21-4c7a-8b2c-07783eef482e)
![Image](https://github.com/user-attachments/assets/7c2bc36d-c7e7-4a7a-98f3-b706c5ed2273)


**Build details:**
VERSION: 8.17.0 BC5
BUILD: 80495
COMMIT: 5c78fb5e4e9b5063bd83ae9bd1e5b32c63f5cc34
Artifact Link: https://staging.elastic.co/8.17.0-a18e6540/downloads/beats/elastic-agent/elastic-agent-8.17.0-windows-x86_64.zip

Hence we are **closing** and marking this issue as **QA:Validated**.

Thanks!
